src/app/review-rating/review-rating.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input } from '@angular/core';
+import { Router } from '@angular/router';
+import { ModalController } from '@ionic/angular';
+import { ReviewRatingService, RatingData } from './review-rating.service';
+
+@Component({
+  selector: 'app-review-rating',
+  templateUrl: './review-rating.component.html',
+  styleUrls: ['./review-rating.component.scss']
+})
+export class ReviewRatingComponent {
+  
+  // Default redirect i.e home page.
+  redirect = ['/'];
+
+  ratingData: RatingData = {
+  	assessment_review_id: null,
+  	rating : 0.5,
+  	comment: '',
+  	tags: []
+  };
+  // variable to control the button text to indicate rating 
+  submitting = false;
```

Comment:
Just a suggestion. I personally think it would be more clear if we call this `isSubmitting`.